incubator/hnc/internal/reconcilers/object.go

```diff
@@ -752,6 +752,9 @@ func (r *ObjectReconciler) syncPropagation(ctx context.Context, log logr.Logger,
 // - Service Account token secrets
 func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstructured.Unstructured, dst string) bool {
 	selector := r.getSelector(log, inst)
+	if selector == nil {
+		selector = r.getTreeSelector(log, inst)
+	}
```

Comment:
Was this the behaviour we defined in the design doc? Ignore the tree selector if there's a non-tree selector? If not, do you think we should change what's in the doc, and if so, why?

Reply:
After reading the design doc again, I think we should 'add' the two selectors together if they both exists. I'll update the logic. However, when the `none` selector exists at the same time with either `select` or `treeSelect` or both of them, do we prioritize the `none` selector?

Reply:
What do you mean by "add" - an AND operation or an OR operation?

If "adding" is hard we could also consider changing the doc. It probably doesn't make a huge difference as most people should only really be using one or the other.

Reply:
OR is easier in this case, do you have a preference if `treeSelector` or `selector` should be prioritized? For me they look equally important

Reply:
When I meant "OR," I mean "a namespace matches if it's in either selector." And "AND" means "matches if it's matched by _both_ selectors." But I think you're describing something different, which is to only look at one or the other. Is that what you meant?

If they're equally important, then maybe it's worth a quick investigation as to how you'd combine them using either an AND or OR operation. I prefer AND, I think, because that has very consistent behaviour with the `none` selector - if `none` is present, nothing ever gets propagated.

Reply:
The current test cases lgtm. BTW after this is addressed, a new test case with both `select` and `treeselect` will be added too, right?

Reply:
Re @adrianludwin: yes I was thinking using one of the selector annotations. Looking into a way to combine both of the selectors.

Re @yiqigao217: Good point, I'll add a test case for this

Reply:
done